### DISCUSSION TITLE
code heat map

### DISCUSSION THREAD

**Discussion by snoopy** [orbitalquark/textadept#558] [Show and tell]
Inspired by the same feature here on Github I've made a code heat map for Textadept.
It allows seeing at a glance if the code is hot or not. The range goes from red (no more than 24 hours old) to dark blue (at least 5 years old).
![heatmap](https://github.com/user-attachments/assets/37df2f39-92f5-4cdf-aea3-58ffdd170a72)

It's based on the output of git-blame. Admittedly, it is pretty gimmicky but might prove useful to quickly discern recent code changes. 
Either way, it was a nice excuse to look into using custom markers. It's also a good example of how easy it is to extend Textadept with whacky things.

**Comment by jxzwp** [comment]
Nice idea. The code for it is here right? 
https://github.com/snoopy/.textadept/blob/a351a4c8a8964748e68b3e0f9d8841a9c95af803/modules/git/init.lua

**Comment by orbitalquark** [comment]
Cool! Thanks for sharing :)
